[Feat] HistoryScreen UI

null4848 opened the pull request:
**Check List :memo:**
- [x] Pull Request 제목 : [유형] 변경 사항 요약
- [x] 설명 & 이슈 번호 작성

**설명 (필수)**

이 PR에서 어떤 변경을 했는지 상세히 설명해주세요.
관련된 Issue 번호를 포함하세요.

content가 없을 경우 나타내기 위해 데이터 변경함

상단에 드롭다운 메뉴 설정

boxes.kt 함수 이용해서 NavyBox 구현
-> content가 없을 경우에만 표시

HomeScreen 코드 참조해서 Card로 기록 뜨게 구현
content가 있을 경우 -> 회색 카드 위에 BlurBackground 
content가 없을 경우 -> 회색 카드
하단에 이름 + 별점은 모두 동일함

LazyColumn이용해서 스크린 전체가 스크롤 가능하게 구현함

boxes -> 이름, 타이틀, 날짜 등등 데이터에서 입력받을 수 있게 수정함
-> 서버랑 연결하고 난 후 수정 가능성 O

홈스크린 참고해서 팝업 UI 설정
title(기록작성), photoadd(사진추가), starratingbar(별점), recordcontent(기록)로 구성
starratingbar에서 0~5 까지 .5 단위로 설정 가능 -> 로그에 별점 저장된 값 출력

fullstar, halfstar, emptystar 아이콘 추가

1. #7 

2. 변경 사항

3. 스크린샷 (선택)
![Screenshot_20250215_220753](https://github.com/user-attachments/assets/eb29e172-4128-48e5-8cf8-9565ea486e8b)
![Screenshot_20250215_220757](https://github.com/user-attachments/assets/7f568608-1512-4f2a-9d2d-35782e98a38f)
![Screenshot_20250215_220811](https://github.com/user-attachments/assets/807cadc3-9287-4063-a578-90e7eaea374f)


<br>

---

**review 참고 사항 (선택)**

추가로 알릴 내용이 있다면 여기에 작성해주세요.

1. 

2.

null4848 committed:
feat: 드롭 다운 메뉴 설정

상단에 드롭 다운 메뉴 설정함

null4848 committed:
feat: 샘플데이터 + UI 중간 저장

샘플 데이터 추가
Card 추가
-> content 데이터에 따른 Card 표시
노트북 변경 이슈로 push함

null4848 committed:
feat: UI 조정

드롭다운 함수 추가
전체적으로 UI 살짝 수정

null4848 committed:
feat: boxes 수정 + 기록하기, Card 수정

기록된 내용 (content)가 없을 시 기록하기 아이템 추가

Card
-> content가 없을 시에는 하단 name + rating
content가 있을시에는 name, adress, content, title, date + 하단 name, rating 표시

전체를 LazyColumn으로 감싸 스크롤 가능하게 구현

별 아이콘 추가

null4848 committed:
Merge branch 'main' into 7-feat-historyscreen-ui

null4848 committed:
cleanup: 의존성 추가 + 클래스 이름 변경

머지 후
Icon, 구글 맵 의존성 추가
데이터 클래스 이름 변경(home screen 이랑 겹침)
-> 비어있는 데이터가 필요해서 일단 이름만 수정함
코드 가독성을 위해 정리

null4848 committed:
feat: 기록하기 버튼 클릭 시 팝업 구현

boxes에서 버튼 클릭시 동장 입력받을 수 있게 변경

홈스크린 참고해서 팝업 UI 설정
title(기록작성), photoadd(사진추가), starratingbar(별점), recordcontent(기록)로 구성

null4848 committed:
feat: icon 추가 + rating 구현

별의 위치를 나누어서 0.0 0.5 1.0
0.5 단위로 설정 가능하게 함

별점을 위해 별 아이콘 추가
fullstar, halfstar, emptystar

의존성 추가

moon0727 commented on app/src/main/java/com/example/golmokstar/ui/screens/HistoryScreen.kt:
```diff
@@ -1,9 +1,552 @@
+}
+
+@Composable
+fun RecordataCard(sampledata: Sampledata) {
```
여기 클릭하면 정보가 뜨는거에요! 기본은 이미지여야합니다


<img width="246" alt="image" src="https://github.com/user-attachments/assets/c7c78078-76ac-46f7-ac5b-40ff7724091e" />

moon0727 commented on app/src/main/java/com/example/golmokstar/ui/screens/HistoryScreen.kt:
```diff
@@ -1,9 +1,552 @@
+    var showDialog by remember { mutableStateOf(false) }
+
+
+    LazyColumn(
```
여기는 Column이 나을 거 같아요!

null4848 replied:
LazyColumn을 Column으로 감싸고 scroll 옵션을 주는 게 불가능한 것 같아요... 수정하다 보니 오류의 늪에 빠지더라고요. 전체 스크롤이 필요해서 LazyColumn으로 감쌌습니다.. 전체 스크롤이 필요하지 않다면 .. 말씀해주신대로 수정하겠습니다

moon0727 replied:
HistoryScreen에
val scrollState = rememberScrollState() 선언 후
Column(HistoryScreen 내부 가장 상위)에
modifier = Modifier.verticalScroll(scrollState)

로 한번 해보시겠어요?

null4848 replied:
보내주신대로 코드로 이미 진행했었는데 영원히 오류를 고치지 못했었습니다 ............

LazyColumn을 Modifier.verticalScroll()이 있는 Column 안에 중첩시키면 높이 제약이 충돌을 일으켜 오류가 발생합니다
Column 안에 LazyColumn을 중첩해서 사용하는 경우, verticalScroll()을 제거하고 LazyColumn을 사용하는 형태로 수정합니다.
LazyColumn을 Column 안에서 빼고 독립적으로 위치시키기 LazyColumn을 Column 안에 넣는 대신, LazyColumn 자체가 스크롤할 수 있는 영역이 되어야 하므로 verticalScroll()을 제거하고 LazyColumn을 Column 안에서 독립적으로 배치합니다.

이거 세개 돌아가면서 해결방안으로 나와서 .. ........ .............................. LazyColumn으로 감싸는게 최선이라고 생각했습니다

moon0727 replied:
아하 그런 과정을 겪으셨군요,,, 알겠습니다!

moon0727 commented on app/src/main/java/com/example/golmokstar/ui/screens/HistoryScreen.kt:
```diff
@@ -1,9 +1,552 @@
+            Spacer(modifier = Modifier.height(25.dp))
+        }
+
+        items(samplehistorydata) { sampledata ->
```
여기서 LazyColumn을 주는게 나을 거 같아요 추가로 재사용되는 컴포넌트에 Spacer를 설정하는 거 보다 부모가 gap을 넣어주는게 나을 거 같습니다! LazyColumn에 아래 스타일을 추가하고 Spacer를 빼는 건 어떨까요?

verticalArrangement = Arrangement.spacedBy(20.dp)

moon0727 commented on app/src/main/java/com/example/golmokstar/ui/screens/HistoryScreen.kt:
```diff
@@ -1,9 +1,552 @@
+
+        items(samplehistorydata) { sampledata ->
+            if (sampledata.content.isEmpty()) {
+                // NavyBox 컴포저블
```
여기 조건에 의문이 있습니다! content뿐만 아니라 rating 존재한다면 기록을 한 것이라 임시적으로 한 거라면 괜찮지만 그게 아니라면 수정이 필요해보입니다.

나중에 server에서 상태를 받아온다고 치고 차라리 sampleData의 상태값을 추가하면 어떨까요? 기록 : Boolean 형식으로 추가해서 조건을 추가하는게 더 좋을 거 같아요!

data class Sampledata(
    val name: String,
    val address: String,
    val imageUrl: String,
    val rating: Double?,
    val title: String,
    val content: String,
    val date: String,
    val history: Boolean
)

null4848 replied:
임시로 그렇게 처리한 거 맞습니다! 기록을 하지 않음 -> content, rating 없음 그래서 임시로 content가 없을 때 기록하기 박스를 뜨게 한 거였어요! 기록 추가하는 거 좋을 것 같습니다 일단 data Class에 기록 여부 추가해두고 조건 추가 구현은 server와 연결 후 하겠습니다

moon0727 commented on app/src/main/java/com/example/golmokstar/ui/screens/HistoryScreen.kt:
```diff
@@ -1,9 +1,552 @@
+            if (sampledata.content.isNotEmpty()) {
+                RecordataCard(sampledata)
+            } else {
+                RecorNodataCard(sampledata)
```
이건 어떤 Card 인가요?

null4848 replied:
기록을 했을 경우와 아닐 경우에 뜨는 Card가 다른 줄 알고 저런식으로 구현했습니다. OptionCard, ClickCard로 수정해두었어요

moon0727 commented on app/src/main/java/com/example/golmokstar/ui/screens/HistoryScreen.kt:
```diff
@@ -1,9 +1,552 @@
+            }
+        }
+    }
+    Spacer(modifier = Modifier.height(20.dp))
```
만약 위에서 말씀드린 LazyColumn에서 verticalArrangement로 고치신다면 여기 삭제하셔도 될겁니다!

moon0727 commented on app/src/main/java/com/example/golmokstar/ui/screens/HistoryScreen.kt:
```diff
@@ -1,9 +1,552 @@
+    ) {
+        Box(modifier = Modifier.fillMaxSize()) {
+            Box(modifier = Modifier.fillMaxSize().background(BlurBackgroundGray))
+
```
Box 두개하신 이유가 있을까요?

null4848 replied:
실수한 것 같습니다! 수정해두었어요 ..

moon0727 commented on app/src/main/java/com/example/golmokstar/ui/screens/HistoryScreen.kt:
```diff
@@ -1,9 +1,552 @@
+            Text(text = sampledata.rating.toString(), color = TextBlack, style = AppTypography.labelMedium)
+        }
+    }
+    Spacer(modifier = Modifier.height(20.dp))
```
만약 위에서 말씀드린 LazyColumn에서 verticalArrangement로 고치신다면 여기 삭제하셔도 될겁니다!

moon0727 commented on app/src/main/java/com/example/golmokstar/ui/screens/HistoryScreen.kt:
```diff
@@ -1,9 +1,552 @@
+                Text(
+                    text = "클릭해서 사진 추가하기",
+                    color = TextDarkGray,
+                    style = AppTypography.bodyLarge
```
bodyMedium으로 해주세요!

moon0727 commented on app/src/main/java/com/example/golmokstar/ui/screens/HistoryScreen.kt:
```diff
@@ -1,9 +1,552 @@
+    }
+
+// Report 다이얼로그 표시
+    if (showDialog) {
```
여기서 조건을 주셨으면 내부에서는 안주셔도 됩니다! Report안에 showDialog 전달 안하셔도 될 거 같아요!

moon0727 commented on app/src/main/java/com/example/golmokstar/ui/screens/HistoryScreen.kt:
```diff
@@ -1,9 +1,552 @@
+        Text("내용 작성", color = TextBlack, style = AppTypography.bodyMedium)
+    }
+
+    OutlinedTextField(
```
여기 border가 두개처럼 보이는데 확인부탁드려요!

null4848 committed:
feat: 코멘트 수정

드롭다운 메뉴 아이콘 수정
-> 의존성 삭제, 아이콘 추가

클릭시 정보 뜨게끔 수정 (기본이 이미지)
-> OptionCard 클릭시 ClickCard

data Class에 기록 여부 추가

OutlinedTextField 외곽선 수정
